ihmc-communication/src/test/java/us/ihmc/communication/ROS2ToolsTest.java

```diff
@@ -165,14 +166,54 @@ public void testLongStringCommunication()
       ThreadTools.sleepForever();
    }
 
+   @Test
+   public void testSharedMemoryROS2Node() throws InterruptedException
+   {
+      AtomicBoolean messageReceived = new AtomicBoolean(false);
+      AtomicBoolean failed = new AtomicBoolean(false);
+      StringBuilder stringToSend = new StringBuilder("Hello World!");
+
+      ROS2Node sharedMemoryNode = new ROS2NodeBuilder().specialTransportMode(SpecialTransportMode.SHARED_MEMORY_ONLY).build(getClass().getSimpleName() + "SharedMemoryNode");
+      ROS2Subscription<std_msgs.msg.dds.String> sharedMemorySubscriber = sharedMemoryNode.createSubscription(ROS2Tools.IHMC_ROOT.withType(std_msgs.msg.dds.String.class), subscriber ->
+      {
+         std_msgs.msg.dds.String message = subscriber.takeNextData();
+         messageReceived.set(true);
+         synchronized (messageReceived)
+         {
+            messageReceived.notify();
+         }
+         assertEquals(stringToSend.toString(), message.getDataAsString());
+         if (!stringToSend.toString().equals(message.getDataAsString()))
+            failed.set(true);
+      });
+
+      ROS2Publisher<std_msgs.msg.dds.String> publisher = sharedMemoryNode.createPublisher(ROS2Tools.IHMC_ROOT.withType(std_msgs.msg.dds.String.class));
+      std_msgs.msg.dds.String messageToSend = new std_msgs.msg.dds.String();
+      messageToSend.setData(stringToSend.toString());
+      publisher.publish(messageToSend);
+
+      synchronized (messageReceived)
+      {
+         messageReceived.wait(1000);
+      }
+      assertTrue(messageReceived.get());
+      assertFalse(failed.get());
+
+      sharedMemorySubscriber.remove();
+   }
+
    @Test
    public void testLoopbackROS2Node() throws InterruptedException, IOException
    {
       AtomicBoolean messageReceived = new AtomicBoolean(false);
       AtomicBoolean failed = new AtomicBoolean(false);
       StringBuilder stringToSend = new StringBuilder("Hello World!");
 
-      ROS2Node loopbackNode = ROS2Tools.createLoopbackROS2Node(PubSubImplementation.FAST_RTPS, getClass().getSimpleName() + "LoopbackNode");
+      int domainId = 111;
```

Comment:
That's your domain. 215 is reserved for unit tests

Reply:
This is not my domain, I set it intentionally. I don't reserve 215 anywhere for unit tests; this is the case in all repos. Typically, the default of 100 is used.

Reply:
That's the one you reserved here and the unit test one is reserved there as well
https://ihmcrobotics.atlassian.net/wiki/spaces/HOWTO/pages/14287727/DDS+Domain+ID+Registry

Reply:
That is private information internal for IHMC. The domain ID effectively shouldn't and doesn't matter in tests. All the workflows are containerized and additionally in this case it's a loopback node.

Reply:
It matters because tests get run locally too, especially when developing them, so they should be isolated as to not mess with robots and sims on the same network.